### PR TITLE
Update webSub module error

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes ##23634

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/websub/src/main/ballerina/src/websub/subscriber_client.bal [hunk 1]
```diff
@@ -123,10 +123,7 @@ function processHubResponse(@untainted string hub, @untainted string mode,
 
     string topic = subscriptionChangeRequest.topic;
     if (response is error) {
-        string errCause = <string> response.detail()?.message;
-        error webSubError = error(WEBSUB_ERROR_CODE, message = "Error occurred for request: Mode[" + mode
-                                        + "] at Hub[" + hub + "] - " + errCause );
-        return webSubError;
+        return error("Error occurred for request: Mode[" + mode+ "] at Hub[" + hub + "] - " + response.message());
```
Comment (chamil321) [inline stdlib/websub/src/main/ballerina/src/websub/subscriber_client.bal:126]:
```suggestion
        return WebSubError("Error occurred for request: Mode[" + mode+ "] at Hub[" + hub + "] - " + response.message());
```

File: stdlib/websub/src/main/ballerina/src/websub/subscriber_client.bal [hunk 2]
```diff
@@ -136,22 +133,18 @@ function processHubResponse(@untainted string hub, @untainted string mode,
                 return invokeClientConnectorOnRedirection(redirected_hub, mode, subscriptionChangeRequest,
                                                             httpClient.config.auth, remainingRedirects - 1);
             }
-            error subscriptionError = error(WEBSUB_ERROR_CODE, message = "Redirection response received for "
-                    + "subscription change request made with followRedirects disabled or after maxCount exceeded: Hub ["
-                    + hub + "], Topic [" + subscriptionChangeRequest.topic + "]");
-            return subscriptionError;
+            return error("Redirection response received for subscription change request made with followRedirects " +
```
Comment (chamil321) [inline stdlib/websub/src/main/ballerina/src/websub/subscriber_client.bal:136]:
```suggestion
            return WebSubError("Redirection response received for subscription change request made with followRedirects " +
```

File: stdlib/websub/src/main/ballerina/src/websub/commons.bal [hunk 3]
```diff
@@ -623,12 +604,11 @@ public type Hub object {
 
         if (stopResult is error) {
             if (stopHubServiceResult is error) {
-                error[] causes = [stopResult, stopHubServiceResult];
-                return error(WEBSUB_ERROR_CODE, causes = causes);
+                return WebSubError("Couldn't stop the started up Ballerina WebSub Hub", stopHubServiceResult);
```
Comment (chamil321) [inline stdlib/websub/src/main/ballerina/src/websub/commons.bal:607]: now can't we pass an array of causes like before?

Comment (kalaiyarasiganeshalingam) [reply stdlib/websub/src/main/ballerina/src/websub/commons.bal:607]: Yes, we can't pass an array of causes now.

**Pull request merged**